hack/lib/test.sh

```diff
@@ -223,3 +228,20 @@ kube::test::if_has_string() {
     return 1
   fi
 }
+
+# Returns true if the required resource is part of supported resources.
+# Expects env vars:
+#   SUPPORTED_RESOURCES: Array of all resources supported by the apiserver. (*)
+#   means it supports all resources.
+#   $1: Name of the resource to be tested.
+kube::test::if_supports_resource() {
```

Comment:
From the build log, almost every check failed (pods, rc, etc.) and thus are skipped https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/38844/pull-kubernetes-unit/10562/build-log.txt

What worries me is that `test-cmd.sh` still passed. How do we make sure this change didn't break / skip some tests by accident?